File: src/domain/repositories/icustomer.repository.ts
import { ICustomer } from '@domain/models';

export interface ICustomerRepository {
  create(customer: ICustomer): Promise<ICustomer>;
  update(customer: ICustomer): Promise<ICustomer>;
  delete(id: string): Promise<void>;
  findAll(): Promise<ICustomer[]>;
  findById(id: string): Promise<ICustomer>;
}

export const ICustomerRepositoryName = 'ICustomerRepository';

File: dotnet/Domain/Repositories/ICustomerRepository.cs
using System;
using System.Collections.Generic;
using dotnet.Domain.Models;

namespace dotnet.Domain.Repositories
{
    public interface ICustomerRepository
    {
        Customer Create(Customer customer);
        Customer Update(Customer customer);
        void Delete(Customer customer);
        List<Customer> FindAll();
        Customer FindById(Guid id);
    }
}
